tornado-api/src/main/java/uk/ac/manchester/tornado/api/types/collections/VectorHalf16.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2024, APT Group, Department of Computer Science,
+ * The University of Manchester.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.manchester.tornado.api.types.collections;
+
+import uk.ac.manchester.tornado.api.types.HalfFloat;
+import uk.ac.manchester.tornado.api.types.arrays.HalfFloatArray;
+import uk.ac.manchester.tornado.api.types.vectors.Half16;
+
+import java.lang.foreign.MemorySegment;
+import java.nio.ShortBuffer;
+
+import static uk.ac.manchester.tornado.api.types.vectors.Half16.add;
+
+public final class VectorHalf16 implements TornadoCollectionInterface<ShortBuffer> {
+
+    public static final Class<VectorHalf16> TYPE = VectorHalf16.class;
+
+    private static final int ELEMENT_SIZE = 16;
```

Comment:
Yes, makes sense. I kept it like that for consistency, because this is how this field is named in all the other vector collection classes. I was thinking to have a separate PR for refactoring all the vector classes at some point, but I can just rename this field for the new classes.